lib/KBox/src/KMessage.h

```diff
@@ -191,6 +196,94 @@ class IMUMessage: public KMessage {
     };
 };
 
+class NAVMessage: public KMessage {
+  private:
+    bool apMode, apHeadingMode, apWaypointMode, apDodgeMode;
+    double currentHeading, targetHeading, courseToWaypoint;
```

Comment:
> We can take out courseToWaypoint if you like. It was there as a placeholder for the next stage in the AP which would be adding the ability to navigate to a waypoint provided from a plotter. 
I understand but I think it's best to keep the code as simple as possible. It will be easy enough to add it when we add support for waypoint. Right now, I think it just makes things more complicated to read and understand.

> Which raises a whole other conversation ... probably should have it elsewhere, so I'll put it on the discussion list ... which is can we get a message back over WiFI from a mobile device running nav software ?

Not yet but I have been working on a big branch that has a ton of changes and that is one of the very important ones. Wifi and host will be able to exchange different type of messages (status of the connection, incoming frames in signalk or nmea format, etc).

> With respect to replacing the mode variables with an ENUM, I thought about that when you suggested it before and started to do it, but then I realized there were issues with that. Let's say we have 4 ENUMs that are mutually exclusive: OFF, HEADINGMODE, DODGEMODE, WAYPOINTMODE. 

What you are building here is a state machine. This is a perfect fit for what you are doing here but the risk when writing them is bugs in the transition from one state to the other (for example, forgetting to update one of the field on the display).
I will take a stab at rewriting this in a slightly different way and hopefully I can show you that it makes things easier to read and understand.